cni/network/invoker_cns_test.go

```diff
@@ -2108,6 +2108,20 @@ func TestShallowCopyIpamAddConfigOptions(t *testing.T) {
 	}
 
 	// shallow copy all ipamAddConfig options
-	res := opts.shallowCopyIpamAddConfigOptions()
-	require.Equal(t, opts.options, res)
+	res1 := opts.shallowCopyIpamAddConfigOptions()
+	require.Equal(t, opts.options, res1)
+
+	// create a second opts for reference representing expected outputs
+	copyRes := res1
+	// modify res1 fields and make sure expectedOpts has not changed
+	res1[network.SNATIPKey] = "100"
+	res1[dockerNetworkOption] = "200"
+	require.Equal(t, copyRes, res1)
```

Comment:
Based on the variable name, is copyRes intended to be a copy of res1's values? What is expectedOpts? If copyRes should not be modified, why are you checking that the two are equal? Shouldn't it be that copyRes should maintain the initial values if it were copied into a seprate map?

Reply:
we should test that after the shallow copy and modifying the shallow copy, the original has not changed